### ISSUE TITLE
Accessibility issue with hashHistory based links

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#334]
# Issue Description
We're currently using react-routers hashHistory feature. This results in links with hrefs formatted with a hash: `http://engineering.cerner.com/terra-core/#/site/arrange`

When running [HTML CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) it reports the following type of errors for all the navigational react-router links.

> This link points to a named anchor "/site/arrange" within the document, but no anchor exists with that name.

## Additional Details (optional)
This should be easy to resolve by replacing react-router's hashHistory module with react-router's browserHistory module. We'll want to test this change on our supported browsers/devices however, support for the browser history API is pretty wide these days: http://caniuse.com/#search=history 

## Issue Type

- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
- Site links don't create accessibility issues

## Current Behavior
- Site links create accessibility issues

**Comment by Matt-Butler** [comment]
If this turns out to be a lot of work, we might want to just do the story to upgrade to react-router v4
